src/panels/Menu/TopMenuBar/TopMenuBar.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { Dispatch, SetStateAction } from 'react';
+import { Button, Group, Kbd, Menu, Text } from '@mantine/core';
+import { modals } from '@mantine/modals';
+
+import { useGetBoolPropertyValue, useOpenSpaceApi } from '@/api/hooks';
+import {
+  AddFileIcon,
+  ConsoleIcon,
+  ExitAppIcon,
+  SaveIcon,
+  VisibilityIcon
+} from '@/icons/icons';
+import { WindowLayoutOptions } from '@/windowmanagement/WindowLayout/WindowLayout';
+
+import { HelpMenu } from './HelpMenu';
+import { NewWindowMenu } from './NewWindowMenu';
+import { TaskBarMenuChoices } from './TaskBarMenuChoices';
+import { MenuWrapper } from './MenuWrapper';
+
+interface TopMenuBarProps {
+  visibleMenuItems: string[];
+  setVisibleMenuItems: Dispatch<SetStateAction<string[]>>;
+  addWindow: (component: React.JSX.Element, options: WindowLayoutOptions) => void;
+}
+
+export function TopMenuBar({
+  visibleMenuItems,
+  setVisibleMenuItems,
+  addWindow
+}: TopMenuBarProps) {
+  const luaApi = useOpenSpaceApi();
+
+  const [isConsoleVisible, setIsConsoleVisible] =
+    useGetBoolPropertyValue('LuaConsole.IsVisible');
+
+  function toggleLuaConsole() {
+    setIsConsoleVisible(!isConsoleVisible);
```

Comment:
usually for toggles i think it is better to use the (old => !old) although here i guess it doesnt really matter since you only manipulate state once